lib/jsonapi/authorization/authorizing_processor.rb

```diff
@@ -216,6 +224,10 @@ def authorize_remove_to_one_relationship
         authorizer.remove_to_one_relationship(source_record, relationship_type)
       end
 
+      def authorize_replace_polymorphic_to_one_relationship
+        raise NotImplementedError
```

Comment:
Good call with just raising `NotImplementedError` here.

We'll need to figure out how to authorize for this polymorphic case before shipping v1.0.0 but that might be better to discuss outside of this PR.